provider/cluster_rosa_classic_data_source.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+type ClusterRosaClassicDataSourceType struct {
+	logger logging.Logger
+}
+
+type ClusterRosaClassicDataSource struct {
+	logger     logging.Logger
+	collection *cmv1.ClustersClient
+}
+
+func (t *ClusterRosaClassicDataSourceType) GetSchema(ctx context.Context) (result tfsdk.Schema,
+	diags diag.Diagnostics) {
+	result = tfsdk.Schema{
+		Description: "OpenShift managed cluster using rosa sts.",
+		Attributes: map[string]tfsdk.Attribute{
+			"name": {
+				Description: "Name of the cluster.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			//future : search with id or name instead of just name
+			"id": {
+				Description: "Unique identifier of the cluster.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"version": {
+				Description: "Identifier of the version of OpenShift, for example 'openshift-v4.1.0'.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"state": {
+				Description: "State of the cluster.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			// to include -> STS Role ARN, Support Role ARN, Instance IAM Roles - Control plane, Instance IAM Roles - Worker, Operator IAM Roles (list), Created,Details Page,OIDC Endpoint URL
+		},
+	}
+	return
+}
+
+func (t *ClusterRosaClassicDataSourceType) NewDataSource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.DataSource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	collection := parent.connection.ClustersMgmt().V1().Clusters()
+
+	// Create the resource:
+	result = &ClusterRosaClassicDataSource{
+		logger:     parent.logger,
+		collection: collection,
+	}
+	return
+}
+
+func (s *ClusterRosaClassicDataSource) Read(ctx context.Context, request tfsdk.ReadDataSourceRequest,
```

Comment:
All clusters have those attributes right? Any reason to call it RosaClassic?

Also, this is confusing I think something like ClusterStateDataSource might be more appropriate.
What do you think?

Reply:
@vamsinm, please open the PR in the new repository - https://github.com/terraform-redhat/terraform-provider-ocm
This repository soon would be archived